ludwig/models/calibrator.py

```diff
@@ -0,0 +1,57 @@
+#! /usr/bin/env python
+# Copyright (c) 2022 Predibase, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+
+import os
+
+import numpy as np
+import torch
+
+from ludwig.globals import MODEL_WEIGHTS_FILE_NAME
+from ludwig.models.ecd import ECD
+from ludwig.models.predictor import Predictor
+
+
+class Calibrator:
+    """Calibrator calibrates the output probabilities of a model."""
+
+    def __init__(self, model: ECD, batch_size: int = 128, skip_save_model=False):
+        self.model = model
+        self.batch_size = batch_size
+        self.skip_save_model = skip_save_model
+
+    def calibration(self, dataset, dataset_name: str, save_path: str):
```

Comment:
Not 100% sure, but it seems to me that in some cases calibration is the act of calibrating outputs, in some cases calibration is the object/function that can calibrate and in some other cases it is the act of training the calibrator.

We probably should be a bit more explicit to avoid confusion.
Wdyt about `train_calibrator`, `get_calibrated_probabilities_from_logits` or something else?
Those may be too verbose, but we can brainstorm a bit on this

Reply:
I like `train_calibrator`, that is clear.  `get_calibrated_probabilities_from_logits` is a good name IMO, where are you thinking that should go (which method would be renamed to this)?  Right now it happens in the PredictModule which just calls the `forward` method of the calibration module.